cavern/src/main/java/org/opencadc/cavern/pkg/CavernPackageRunner.java

```diff
@@ -0,0 +1,297 @@
+     * @param packageParentPath - Path to root of the package being built
+     * @param piList - list of PackageItem being added to
+     */
+    private void addToPkgList(Path targetPath, Path packageParentPath, List<PackageItem> piList) {
+        log.debug("adding path to list: " + targetPath.toString());
+        try {
+            // Generate a file:// URL using the targetPath passed in
+            // Assumption here is canfar_root is well known, so there isn't a need
+            // to compare the path parts - it has been added to anything in the ManifestList,
+            // so subpath using count of parts is safe here
+            URL fileURL = new URL("file://" + targetPath.toString());
+            log.debug("file uri: " + fileURL.toString());
+
+            Path relativePath;
+            int parentPathLength = packageParentPath.getNameCount();
+            log.debug("parent path length: " + parentPathLength);
+
+            int targetPathLength = targetPath.getNameCount();
+            log.debug("targetPath part count: " + targetPathLength);
+            relativePath = targetPath.subpath(parentPathLength, targetPathLength);
+
+            String relativePathStr = "/" + relativePath.toString();
+            log.debug("relativePath: " + relativePathStr);
+
+            log.debug("adding to PackageItem list");
+
+            PackageItem pi = new PackageItem(fileURL, relativePathStr);
+            piList.add(pi);
+        }  catch (MalformedURLException mue) {
+            log.info("Malformed URL: " + targetPath.toString() + " - skipping...");
+        }
+    }
+
+    /**
+     * FileVisitor implementation that adds files visited to the PackageItem
+     * list owned by CavernPackageRunner
+     */
+    private static class ManifestVisitor implements FileVisitor<Path> {
+
+        private Path root;
+        private List<Path> manifestList;
+
+        ManifestVisitor(Path root) {
+            this.root = root;
+            manifestList = new ArrayList<>();
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path t, BasicFileAttributes bfa)
+            throws IOException {
+            log.debug("ManifestVisitor: pre-visit directory: " + t);
+            return FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path t, BasicFileAttributes bfa)
+            throws IOException {
+            log.debug("ManifestVisitor: visiting file: " + t);
+
+            // Add the foot to the name of the file being visited
+            Path file = root.relativize(t);
+            log.debug("ManifestVisitor: adding file to manifest list: " + file);
+            manifestList.add(file);
+            return FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFileFailed(Path t, IOException ioe)
+            throws IOException {
+            return FileVisitResult.CONTINUE;
```

Comment:
Should the error be logged here?  (and in the method below?)

Reply:
logged this one as an 'info' statement. Added debug statements in the postDirectoryVisit - IOException there *could* be not null, but it's a function passed through after each pass through a directory. Added a debug statement.

Reply:
Should only be done if the exception is not null. (maybe it's always non-null for this?). Anyhow, it should be ERROR level.  INFO level is only used for the start and end log messages.

Reply:
done